linalg/hypre.cpp

```diff
@@ -3255,6 +3256,20 @@ HypreParMatrix * HypreParMatrixFromBlocks(Array2D<HypreParMatrix*> &blocks,
    }
 }
 
+HypreParMatrix * HypreParMatrixFromBlocks(Array2D<HypreParMatrix*> &blocks,
+                                          Array2D<double> *blockCoeff)
+{
+   Array2D<const HypreParMatrix*> constBlocks(blocks.NumRows(), blocks.NumCols());
+   for (int i = 0; i < blocks.NumRows(); ++i)
+   {
+      for (int j = 0; j < blocks.NumCols(); ++j)
+      {
+         constBlocks(i, j) = blocks(i, j);
```

Comment:
Unfortunately, we cannot do it better than this right now, because `Array2D` does not have a converting constructor unlike `Array`. Does anyone want to write one? 🙂 It could be a follow-up also of #4136 . If no one, I will do that in a separate PR. Anyways, I am approving this one 👍